cucumber-android/src/main/java/cucumber/runtime/android/CucumberJUnitRunner.java

```diff
@@ -0,0 +1,130 @@
+package cucumber.runtime.android;
+
+import org.junit.runner.Description;
+import org.junit.runner.Result;
+import org.junit.runner.Runner;
+import org.junit.runner.manipulation.Filter;
+import org.junit.runner.manipulation.Filterable;
+import org.junit.runner.manipulation.NoTestsRemainException;
+import org.junit.runner.notification.RunNotifier;
+
+import android.support.test.InstrumentationRegistry;
+import android.util.Pair;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.IdentityHashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import cucumber.api.android.CucumberAndroidJUnitRunner;
+import cucumber.runtime.Utils;
+import gherkin.events.PickleEvent;
+
+public class CucumberJUnitRunner extends Runner implements Filterable {
+
+    private final CucumberAndroidJUnitRunner instrumentationRunner;
+
+    private final CucumberExecutor cucumberExecutor;
+
+    private final List<PickleEvent> pickleEvents;
+
+    public CucumberJUnitRunner(Class testClass) {
+        instrumentationRunner = (CucumberAndroidJUnitRunner) InstrumentationRegistry
+                .getInstrumentation();
+        cucumberExecutor = new CucumberExecutor(new Arguments(instrumentationRunner.getArguments()),
```

Comment:
`Bundle` arguments are stored in `CucumberAndroidJUnitRunner` for backward compatibility with Cucumbers tags/configuring